### PR TITLE
Smarter Seeding for Better Reproduceability

### PR DESCRIPTION
**Pull request by JHiltbrand**
--allow user to seed modules with specified seed
--in any case, store seed in config dictionary for use later
--try some Keras and tensorflow clear/reset commands at the very beginning
--set some env variables to help with determinism and randomness.

After testing with all of these things in place, it is possible to exactly reproduce a training (down to the fourth decimal for 3 epochs), however, minor fluctuations are still observed. Overall, the total variability, on average, is much more constrained

### CODE REVIEW

**Comment by JHiltbrand** [comment]
Some other unrelated, but benign and minor changes to `train.py` as well.

**Comment by JHiltbrand** [comment]
Also, seed numpy in `DataGetter.py` with the same seed as used everywhere else
